### ISSUE TITLE
decide on where to include team/model name information. 

### ISSUE THREAD

**Issue by nickreich** [Infectious-Disease-Modeling-Hubs/hubDocs#2]
From @elray1 

> Note that the proposal [i.e. the current documentation] here does not include the team and model name in the file name. To discuss:
> - advantages:
> - eliminates redundant information
> - means we do not need to do any regular expression parsing of file names to extract information about team name and round id
> - disadvantages:
> - removes an opportunity to validate that submissions are in the correct folder (e.g., team didn't mix up forecasts from different models they are responsible for, team didn't drop their forecasts into someone else's folder)

There have been ongoing discussions as well about whether or not model name should be included in the file explicitly as well.

**Comment by nickreich** [comment]
relevant to content e.g. [here](https://github.com/Infectious-Disease-Modeling-Hubs/hubDocs/blob/66423fc9131ab71910a6b3912fb4f630d0f50907/docs/source/format/hub-structure.md?plain=1#L15)

**Comment by elray1** [comment]
Following up on this -- more generally, I think we should decide on how both the model name and the round id are represented (these are the two items that are encoded in file paths). Recall that in our current proposal at the link Nick provided just above, the model outputs are stored in files with the following organizational structure:

```
* model-output directory: e.g., Forecast data or scenario projections produced by participating models/teams
   1. team1-modela
      * <round-id1>.csv[k] (or parquet, etc)
      * <round-id2>.csv (or parquet, etc)
   2. team1-modelb
      * <round-id1>.csv (or parquet, etc)
   3. team2-modela
      * <round-id1>.csv (or parquet, etc)
```

I propose the following as a guiding principle: model output submission files should contain enough data (within the file) to uniquely identify different forecasts.

I think that this suggests the following rules about representation of models and round ids:

1. In all hubs, submission files should contain columns that uniquely identify the team/model that generated the model outputs.
    - This would mean that if we load forecasts/projections from two different models and "row bind" them, it will be guaranteed that there are no duplicate rows from different models. Additionally, note that we will often need the model in common workflows, e.g. plotting forecasts from different models or building trained ensembles. We will therefore typically want to have the model in data objects with model outputs, and having them stored as data in the submission files means that we don't need to insert special logic to handle this when loading model outputs.
    - I see two options for what this could look like. I propose that we pick one of them and require all hubs to use this standard. I prefer option 1 since it's easier to paste two columns together than to split them apart, but I could easily be convinced to go the other way too.
        1. Separate columns for the `team_abbr` and the `model_abbr`.
        2. A single `team_model_abbr` column where the value is of the form `<team_abbr>-<model_abbr>`
    - In either case, we could validate that the file is in the right folder since the folder name should match the `team_abbr` and `model_abbr` in the file.

2. It is *not* required for hubs to put the `round id` as a column in the submission file, but it *is* required that there are not multiple rounds that have the same combinations of values for all task id variables within one hub.
    - This would mean that if we load forecasts/projections from two different rounds and "row bind" them, it will be guaranteed that there are no duplicate rows from different rounds.
    - This would also mean that if we ever needed to reconstruct the round id corresponding to a particular model output row, we could in principle do so by looking at the task id variable/value combinations for different rounds and identifying which round had the combination that shows up in the model output row we're inspecting.
    - In practice, forecast hubs that use `"round_id_from_variable": true` will have the variable corresponding to the round id in submission files anyways.
    - Hubs that use `"round_id_from_variable": false` can always list the round id as a task id variable at their discretion.

**Comment by harryhoch** [comment]
@elray1, this makes sense. 

I think it might be good to state some explicit design principles.  In this case, you're sacrificing file size to gain some unambiguous clarity.  this is a reasonable tradeoff.    I can imagine an effort that might make a different choice: I worked on a project once where metadata files was preserved by wrapping the entire data file in an XML block. This keeps the metadata close at hand, but makes life difficult for human access and editing.   Talking about these decisions in terms of tradeoffs between parsimony/file size, data fidelity, and ease of use might be one way to frame these discussions.

**Comment by elray1** [comment]
Summing up a few misc. comments from in person conversation:

comments r.e. model/team name:
- after loading model outputs into data analysis environment (R/python), we're going to need to have columns for the team/model in data objects with the forecasts or projections for common workflows including scoring, plotting, and weighted ensembling. at minimum, we'll need a standardized format that includes these columns.
- some people in favor of including model in data files, and others preferred not to require hubs to introduce
- Q. from Harry: "why not insist upon including the team name (or abbreviation) in the file name along with the round name?"
    - Previous argument from Hugo: desire not to have to parse file names.

comments r.e. round id
- Mike J gave an example of a WNV forecast hub where the targets were the same across multiple rounds, except for the due date

**Comment by harryhoch** [comment]
Clarification. My suggestion was intended as including the team name _in addition to_ the round name.  Including both in both the file name and the file would sacrifice brevity while providing greater flexibility and clarity.

**Comment by elray1** [comment]
Here's another proposal for handling of model names, feeding off of comments in our meeting and a little more sidebar discussion later:

- we define a standard data object format that includes the information about the model/team as columns (TBD whether this is one or two columns, but this will be a standardized format that is consistent across all hubs). Functions like `hubUtils::load_predictions` will produce as output data frames that are assured to have these columns after reading predictions into an environment like an R or python session.
- we support multiple options for back-end storage of these data. A particular hub may pick one of these options, and a task for a `load_forecasts` function will be to standardize the representation of the data after they are loaded in:
    - The original proposal, where the model/team name is encoded in file paths, but they do not necessarily appear within the file. (Need to do some more reading to be sure I understand, but note that this may be [supported already in parquet](https://arrow.apache.org/docs/r/reference/open_dataset.html#partitioning-1))
    - The alternate proposal I made above where data files actually have model/team abbreviations as columns
    - Possibly other schemes that could be introduced later.

**Comment by nickreich** [comment]
1. I vote yes, to move ahead.
2. I would vote for two columns.
3. I concur with Evan's suggestion about having an "origin_date" column.

**Comment by LucieContamin** [comment]
I agree too.
1. Yes, I think this proposal answers all our questions and represents a good consensus
2. I would also vote for 2 columns (`team_abbr` and `model_abbr` sounds good)
3. I agree with your suggestion to add a column storing this information but I think we need a precise description of all the differents columns possible for date information, to avoid confusion and maybe avoid having 2 hubs with the same column name but with different meaning for the value (or at least not without good documentation)

**Comment by mzorn-58** [comment]
Unclear what is to be done. Should the following be added to documentation?  Any specifics about where to add it?

- File name should include team and model name
- column in model output data should include team and model name as single variable

**Comment by nickreich** [comment]
@annakrystalli This is prompting a question for me. Should the recommendation be to never include a model-id column in the submitted file since it will always be implicit in the directory name? And if it were included then `collect()` might include it twice, right? or is there an allowance for additional columns that are not specified in the schema and that could be removed prior to `collect()`ion?

**Comment by elray1** [comment]
r.e. question just above -- if we want to ensure that submitted model output files are in the right folder, we may need this as a column in the submitted file?

**Comment by elray1** [comment]
if we have decided on model id as the standard here (which concatenates a team abbreviation and model abbreviation), maybe we need to update the `Model metadata` documentation to reflect this.

**Comment by elray1** [comment]
ok, thanks. sounds like it's easier not to put this into the file name, at least for now.

**Comment by annakrystalli** [comment]
@elray1 , you mean: "...easier not to put this into the FILE." right? We still want this information in the file name so we can check a file is being submitted to the correct folder?

**Comment by elray1** [comment]
yes, sorry, and thanks :)

**Comment by elray1** [comment]
I suggest that we make one small updates and then close this issue: on [this page](https://hubdocs.readthedocs.io/en/latest/user-guide/model-metadata.html#template-metadata-schema-file), when describing the template model metadata schema file, let's say that a hub's model metadata schema file should incude both of the fields `team_abbr` and `model_abbr`, or the single combined field `model_id`.

**Comment by micokoch** [comment]
Addressed in PR #101
